test/e2e/maxconn_test.go

```diff
@@ -0,0 +1,164 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestTunableMaxConnectionsValidValues(t *testing.T) {
+	updateMaxConnections := func(t *testing.T, client client.Client, timeout time.Duration, maxConnections int32, c *operatorv1.IngressController) error {
```

Comment:
Seems you only use `c` as a way to get the name of the object and then as a place to store the object that you get from the API, which is subtle.  What do you think about replacing the `*operatorv1.IngressController` parameter with a `types.NamespacedName` parameter and using a local variable to store the object from the API?

Reply:
Would it make more sense just to use `kclient` and `time.Minute` instead of parameterizing the client and timeout?

Reply:
I want the caller to specify how long to wait for. And I'd rather not have a mismatch of globals and discrete parameter values.

Reply:
> types.NamespacedName

Done.